### PR TITLE
Fix path transformation in data_deps()

### LINKED ISSUE #1614: BS4 leads to src_path starting with `%3A` instead of `:` in pkgdown_1.6.1.9001
Thanks for the great package and your work on it!

Today, I run into some problems trying out BS4 on Windows (while it works for `bootstrap = 3`).

It seems that the function `data_deps` return (at least for parameter `depth = 0L`) non-valid file paths of the form 
`C%3AUsers/mtkerber/AppData/Local/Temp/Rtmpicht2I/testpkg2/docs/deps/jquery-3.5.1/jquery-3.5.1.min.js`

Minimal reprex:

``` r
library(magrittr)
#> Warning: Paket 'magrittr' wurde unter R Version 4.0.3 erstellt
library(pkgdown)
library(sass)
#> Warning: Paket 'sass' wurde unter R Version 4.0.3 erstellt
library(bslib)
#> Warning: Paket 'bslib' wurde unter R Version 4.0.4 erstellt
options(pkgdown.internet = FALSE)

pkg_path <- fs::path(fs::path_temp(), "testpkg2")
usethis::create_package(path = pkg_path, open = FALSE)
#> v Creating 'C:/Users/mtkerber/AppData/Local/Temp/Rtmp2pFLgn/testpkg2/'
#> v Setting active project to 'C:/Users/mtkerber/AppData/Local/Temp/Rtmp2pFLgn/testpkg2'
#> v Creating 'R/'
#> v Writing 'DESCRIPTION'
#> Package: testpkg2
#> Title: What the Package Does (One Line, Title Case)
#> Version: 0.0.0.9000
#> Authors@R (parsed):
#>     * First Last <first.last@example.com> [aut, cre] (YOUR-ORCID-ID)
#> Description: What the package does (one paragraph).
#> License: `use_mit_license()`, `use_gpl3_license()` or friends to pick a
#>     license
#> Encoding: UTF-8
#> LazyData: true
#> Roxygen: list(markdown = TRUE)
#> RoxygenNote: 7.1.1
#> v Writing 'NAMESPACE'
#> v Setting active project to '<no active project>'

# does not work - using bootstrap 4
pkgdown::build_site(pkg_path, override = list(template = list(bootstrap = 4)))
#> -- Installing package into temporary library -----------------------------------
#> == Building pkgdown site =======================================================
#> Reading from: 'C:/Users/mtkerber/AppData/Local/Temp/Rtmp2pFLgn/testpkg2'
#> Writing to:   'C:/Users/mtkerber/AppData/Local/Temp/Rtmp2pFLgn/testpkg2/docs'
#> -- Initialising site -----------------------------------------------------------
#> Copying '../../../../../../../Program Files/R/R-3.6.0/library/pkgdown/assets/BS4/docsearch.css' to 'docsearch.css'
#> Copying '../../../../../../../Program Files/R/R-3.6.0/library/pkgdown/assets/BS4/docsearch.js' to 'docsearch.js'
#> Copying '../../../../../../../Program Files/R/R-3.6.0/library/pkgdown/assets/BS4/link.svg' to 'link.svg'
#> Copying '../../../../../../../Program Files/R/R-3.6.0/library/pkgdown/assets/BS4/pkgdown.css' to 'pkgdown.css'
#> Copying '../../../../../../../Program Files/R/R-3.6.0/library/pkgdown/assets/BS4/pkgdown.js' to 'pkgdown.js'
#> Copying '../../../../../../../Program Files/R/R-3.6.0/library/pkgdown/assets/BS4/syntax-highlighting.css' to 'syntax-highlighting.css'
#> Writing '404.html'
#> -- Building home ---------------------------------------------------------------
#> Writing 'authors.html'
#> -- Building function reference -------------------------------------------------
#> Writing 'reference/index.html'
#> == DONE ========================================================================
src_bs4 <- rvest::read_html(fs::path(pkg_path, "docs", "index.html")) %>% 
  rvest::html_element("script") %>% 
  rvest::html_attr("src")
src_bs4
#> [1] "C%3AUsers/mtkerber/AppData/Local/Temp/Rtmp2pFLgn/testpkg2/docs/deps/jquery-3.5.1/jquery-3.5.1.min.js"

sessionInfo()
#> R version 4.0.1 (2020-06-06)
#> Platform: x86_64-w64-mingw32/x64 (64-bit)
#> Running under: Windows 10 x64 (build 18363)
#> 
#> Matrix products: default
#> 
#> locale:
#> [1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252   
#> [3] LC_MONETARY=German_Germany.1252 LC_NUMERIC=C                   
#> [5] LC_TIME=German_Germany.1252    
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> other attached packages:
#> [1] bslib_0.2.4        sass_0.3.1         pkgdown_1.6.1.9001 magrittr_2.0.1    
#> 
#> loaded via a namespace (and not attached):
#>  [1] Rcpp_1.0.6        pillar_1.5.1      compiler_4.0.1    jquerylib_0.1.3  
#>  [5] highr_0.8         tools_4.0.1       digest_0.6.27     jsonlite_1.7.2   
#>  [9] evaluate_0.14     memoise_2.0.0     lifecycle_1.0.0   tibble_3.1.0     
#> [13] pkgconfig_2.0.3   rlang_0.4.10      reprex_1.0.0      rstudioapi_0.13  
#> [17] cli_2.3.1         yaml_2.2.1        xfun_0.22         fastmap_1.1.0    
#> [21] httr_1.4.2        xml2_1.3.2        withr_2.4.1       styler_1.4.1     
#> [25] stringr_1.4.0     knitr_1.31        roxygen2_7.1.1    desc_1.3.0       
#> [29] fs_1.5.0          vctrs_0.3.6       rprojroot_2.0.2   glue_1.4.2       
#> [33] R6_2.5.0          processx_3.5.0    fansi_0.4.2       rmarkdown_2.7    
#> [37] selectr_0.4-2     callr_3.6.0       purrr_0.3.4       ps_1.6.0         
#> [41] backports_1.2.1   ellipsis_0.3.1    htmltools_0.5.1.1 usethis_2.0.1    
#> [45] rvest_1.0.0       assertthat_0.2.1  utf8_1.2.1        stringi_1.5.3    
#> [49] cachem_1.0.4      crayon_1.4.1
```

<sup>Created on 2021-04-07 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

### PR DESCRIPTION
**Pull request by maelle**
Fix #1614

### CODE REVIEW

**Comment by maelle** [comment]
not fixed but reproduced at the moment.

**Comment by hadley** [comment]
This looks like a url escaping problem?

**Comment by maelle** [comment]
`"../../D%3A/a/pkgdown/pkgdown/docs/dev/deps/jquery-3.5.1/jquery-3.5.1.min.js"` should be `deps/jquery-3.5.1/jquery-3.5.1.min.js` which is why I think the problem might be in https://github.com/r-lib/pkgdown/blob/87b7fd66b737c44775e4b678c821bde014ad2065/R/render.r#L433

**Comment by maelle** [comment]
And then yes `htmltools::renderDependencies()` encodes the path (path that is wrong).

**Comment by maelle** [comment]
Or maybe htmtltools does the encoding earlier than I thought, I'll look.

**Comment by maelle** [comment]
Once the fix is approved I'll remove the edits of the GHA workflow file.

File: R/render.r [hunk 1]
```diff
@@ -432,7 +432,7 @@ data_deps <- function(pkg, depth) {
   # Function needed for indicating where that deps folder is compared to here
   transform_path <- function(x) {
 
-    x <- gsub(pkg$dst_path, "", x)
+    x <- sub(htmltools::urlEncodePath(pkg$dst_path), "", htmltools::urlEncodePath(x))
```
Comment (hadley) [inline R/render.r:435]: Can you explain why this works?

Comment (maelle) [reply R/render.r]: Which also makes me realize I was using the function one time too many.

**Review by hadley — APPROVED** [review]

**Pull request merged**